mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -526,7 +526,7 @@ class LogisticRegression @Since("1.2.0") (
       case None => histogram.length
     }
 
-    val isMultinomial = $(family) match {
+    val isMultinomial = $(family).toLowerCase(Locale.ROOT) match {
```

Comment:
As a general practice, I would recommend moving the `.toLowerCase(Locale.ROOT)` into the setter. Then we don't need to invoke the `.toLowerCase(Locale.ROOT)` multiple times in the code. (here it happens to be once). And we can always assume the $(family) has predictable values in the code.